pkg/cmd/provider/install.go

```diff
@@ -126,11 +110,33 @@ func convertToDTO(manifest *manager.ProvidersManifest) []apiclient.Provider {
 	return pluginList
 }
 
-func convertToStringMap(downloadUrls map[os.OperatingSystem]string) map[string]string {
+func ConvertToStringMap(downloadUrls map[os.OperatingSystem]string) map[string]string {
```

Comment:
Just a thought, consider perhaps renaming this function to `ConvertOSToStringMap` or something along those lines since it has a specific type (`os.OperatingSystem`) for conversion and name (at least for me) indicates that it converts generally (no matter the underlying types).
Another thought is to create a non-type-specific util function which accepts interface as conversion type and the conversion is done on site where needed?

Reply:
I believe the best course of action would be to have `type OperatingSystemMap (or something) map[os.OperatingSystem]string` and then we define a method on that type for converting to `map[string]string`.

But I agree with your point generally.